File: ruby/SuppliesPackage.rb
# Representa a un paquete de suministros para una estación espacial. Puede
# contener armamento, combustible y/o energía para los escudos.

module Deepspace
    class SuppliesPackage
        def initialize(initAmmoPower, initFuelUnits, initShieldPower)
            @ammoPower = initAmmoPower
            @fuelUnits = initFuelUnits
            @shieldPower = initShieldPower
        end

        # Constructor de copia
        def self.newCopy(other)
            new(other.ammoPower, other.fuelUnits, other.shieldPower)
        end

        def ammoPower
            @ammoPower
        end

        def fuelUnits
            @fuelUnits
        end

        def shieldPower
            @shieldPower
        end

        def to_s
            "ammoPower: #@ammoPower, fuelUnits: #@fuelUnits, shieldPower: #@shieldPower";
        end
    end
end

File: java/DeepSpace/src/deepspace/SuppliesPackage.java
package deepspace;


// Representa a un paquete de suministros para una estación espacial. Puede
// contener armamento, combustible y/o energía para los escudos.

class SuppliesPackage {
    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
    
    SuppliesPackage(float ammoPower, float fuelUnits, float shieldPower){
        this.ammoPower = ammoPower;
        this.fuelUnits = fuelUnits;
        this.shieldPower = shieldPower;
    }
    
    SuppliesPackage(SuppliesPackage other){
        this.ammoPower = other.ammoPower;
        this.fuelUnits = other.fuelUnits;
        this.shieldPower = other.shieldPower;
    }
    
    public float getAmmoPower(){
        return ammoPower;
    }
    
    public float getFuelUnits(){
        return fuelUnits;
    }
    
    public float getShieldPower(){
        return shieldPower;
    }
    
    public String toString(){
        return "ammoPower: " + ammoPower +
                ", fuelUnits: " + fuelUnits +
                ", shieldPower: " + shieldPower;
    }
}
